code/modules/mob/abstract/new_player/menu.dm

```diff
@@ -48,247 +48,188 @@
-		for(var/screen in known_icon_states)
-			if(!(screen in SSatlas.current_map.lobby_screens))
-				SSatlas.current_map.lobby_screens += screen
-	icon = SSatlas.current_map.lobby_icon
-
-	if(length(SSatlas.current_map.lobby_screens))
-		if(SSatlas.current_map.lobby_transitions && isnum(SSatlas.current_map.lobby_transitions))
-			icon_state = SSatlas.current_map.lobby_screens[lobby_index]
-			if(!MC_RUNNING())
-				spawn(SSatlas.current_map.lobby_transitions)
-					Update()
-			else
-				refresh_timer_id = addtimer(CALLBACK(src, PROC_REF(Update)), SSatlas.current_map.lobby_transitions, TIMER_UNIQUE | TIMER_CLIENT_TIME | TIMER_OVERRIDE | TIMER_STOPPABLE)
-		else
-			icon_state = pick(SSatlas.current_map.lobby_screens)
-	else //This should basically never happen.
-		crash_with("No lobby screens found!")
-
 	. = ..()
 
-/atom/movable/screen/new_player/title/set_sector_things()
-	return
+	setup_icon() //THIS CAN SLEEP AND MUST NOT BE WAITED FOR
 
-/atom/movable/screen/new_player/title/proc/Update()
-	SHOULD_NOT_SLEEP(TRUE)
+/**
+ * Sets up the icon for the title screen, wait until SSAtlas made them for us then setup the update cycle after picking one
+ */
+/atom/movable/screen/new_player/title/proc/setup_icon()
+	set waitfor = FALSE
+	UNTIL((SSatlas.lobby_icons))
+
+	icon = pick(SSatlas.lobby_icons)
+
+	addtimer(CALLBACK(src, PROC_REF(update_icon)), SSatlas.current_map.lobby_transitions, TIMER_UNIQUE | TIMER_OVERRIDE)
+
+/atom/movable/screen/new_player/title/update_icon()
+	..()
 
 	if(QDELETED(src))
 		return
 
-	if(!SSatlas.current_map.lobby_transitions && SSatlas.current_sector.sector_lobby_transitions)
-		return
 	if(!istype(hud) || !isnewplayer(hud.mymob))
 		return
-	lobby_index += 1
-	if (lobby_index > length(SSatlas.current_map.lobby_screens))
-		lobby_index = 1
-	animate(src, alpha = 0, time = 1 SECOND)
-	animate(alpha = 255, icon_state = SSatlas.current_map.lobby_screens[lobby_index], time = 1 SECOND)
-	refresh_timer_id = addtimer(CALLBACK(src, PROC_REF(Update)), SSatlas.current_map.lobby_transitions, TIMER_UNIQUE | TIMER_CLIENT_TIME | TIMER_OVERRIDE | TIMER_STOPPABLE)
-
-/atom/movable/screen/new_player/selection
+
+	//No point fading if there's only one icon, but we keep the update running in case admins want to add icons on the fly
+	if(length(SSatlas.lobby_icons) >= 2)
+		//Advance to the next icon
+		lobby_icons_index = max(++lobby_icons_index % length(SSatlas.lobby_icons), 1)
+
+		animate(src, alpha = 0, time = 1 SECOND)
+
+		animate(alpha = 255, icon = SSatlas.lobby_icons[lobby_icons_index], time = 1 SECOND)
+
+	addtimer(CALLBACK(src, PROC_REF(update_icon)), SSatlas.current_map.lobby_transitions, TIMER_UNIQUE | TIMER_OVERRIDE)
+
+
+/**
+ * # Selection screen
+ *
+ * Lobby clickable buttons
+ */
+ABSTRACT_TYPE(/atom/movable/screen/new_player/selection)
+	icon_state = null
+	mouse_opacity = MOUSE_OPACITY_OPAQUE
 	var/click_sound = 'sound/effects/menu_click.ogg'
-	var/hud_arrow
+	var/does_matrix_scale = TRUE
+	var/uses_hud_arrow = TRUE
```

Comment:
I think they're fairly clear on what they do but it would be optimal if they were documented